bootstrap/shell/shell.go

```diff
@@ -240,10 +250,22 @@ func (s *Shell) RunAndCapture(command string, arg ...string) (string, error) {
 	return strings.TrimSpace(b.String()), nil
 }
 
+// injectTraceCtx adds tracing information to the given env vars to support
+// distributed tracing across jobs/builds.
+func (s *Shell) injectTraceCtx(ctx context.Context, env *env.Environment) {
+	span := opentracing.SpanFromContext(ctx)
+	// Not all shell runs will have tracing (nor do they really need to).
+	if span == nil {
+		return
+	}
+	// Intentionally ignoring error. Logging will spam agent logs.
+	_ = opentracing.GlobalTracer().Inject(span.Context(), opentracing.TextMap, env.ToMap())
```

Comment:
Could this be ignored _unless_ the buildkite-agent has had the `--debug` flag specified, please?

Reply:
Oh - yeah, log it, definitely. Sorry, could have been clearer!